src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -31,6 +34,118 @@ const SHA224: i64 = 224;
 const SHA256: i64 = 256;
 const SHA384: i64 = 384;
 const SHA512: i64 = 512;
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum AESMode {
+    Aes128Ecb,
+    Aes192Ecb,
+    Aes256Ecb,
+    Aes128Cbc,
+    Aes192Cbc,
+    Aes256Cbc,
+}
+
+impl AESMode {
+    pub fn key_size(self) -> aes::KeySize {
+        match self {
+            AESMode::Aes128Ecb | AESMode::Aes128Cbc => aes::KeySize::KeySize128,
+            AESMode::Aes192Ecb | AESMode::Aes192Cbc => aes::KeySize::KeySize192,
+            AESMode::Aes256Ecb | AESMode::Aes256Cbc => aes::KeySize::KeySize256,
+        }
+    }
+
+    pub fn from(mode: &[u8]) -> Option<AESMode> {
+        let mode = match String::from_utf8(mode.to_vec()) {
+            Ok(v) => v,
+            Err(_) => {
+                return None;
+            }
+        };
+        match mode.as_ref() {
+            "aes-128-ecb" => Some(AESMode::Aes128Ecb),
+            "aes-192-ecb" => Some(AESMode::Aes192Ecb),
+            "aes-256-ecb" => Some(AESMode::Aes256Ecb),
+            "aes-128-cbc" => Some(AESMode::Aes128Cbc),
+            "aes-192-cbc" => Some(AESMode::Aes192Cbc),
+            "aes-256-cbc" => Some(AESMode::Aes256Cbc),
+            _ => None,
+        }
+    }
+
+    fn derive_key_mysql(input: &[u8], block_size: usize) -> Vec<u8> {
+        let mut output = vec![0; block_size];
+        let mut idx = 0;
+        for k in input {
+            if idx == block_size {
+                idx = 0;
+            }
+            output[idx] ^= k;
+            idx += 1;
+        }
+        output
+    }
+
+    pub fn build_encryptor(self, key: &[u8], iv: Option<&[u8]>) -> Result<Box<Encryptor>> {
+        let key_size = match self {
+            AESMode::Aes128Ecb | AESMode::Aes128Cbc => 16,
```

Comment:
Why not use `self.key_size()`?

Reply:
self.key_size() return aes::KeySize，but this need a i32